src/core/Akka/Serialization/Serialization.cs

```diff
@@ -337,8 +337,8 @@ public void AddSerializer(Serializer serializer)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void AddSerializer(string name, Serializer serializer)
         {
-            _serializersById.Add(serializer.Identifier, serializer);
-            _serializersByName.Add(name, serializer);
+            _serializersById[serializer.Identifier] = serializer;
```

Comment:
It is likely going to introduce some breaking changes? If there are multiple serializer with same name, the last inserted will override the previous one - the effect is that the last one will win if there are two different serializer with same name.

```csharp
__serializersById = new Dictionary<string, List<Type>>();
``` 
I don't know much about `Serialization.cs`, but if  there are no other way to solve that problem, is that doable, is it "legal"?

Reply:
That is the point, to enable people to override serializers

Reply:
Okay. Will the custom serializer be registered last after the default serializers have been registered?

Reply:
I would like to add a warning when a serializer is being overridden, to help protect against accidental footguns. We should check here to see if this key is already contained in the collection or not and then log "you are replacing X with Y - did you mean to do this?" as a WARNING loglevel